src/clld_markdown_plugin/__init__.py

```diff
@@ -37,14 +37,23 @@ def full_spec(spec):
         'renderer_map': {
             "ExampleTable": render_ex,
         },
+        "extensions": [
+            "markdown.extensions.fenced_code",
+            "markdown.extensions.md_in_html",
+            "markdown.extensions.tables",
+            "markdown.extensions.attr_list",
+            "markdown.extensions.footnotes",
+        ]
     }
     custom = custom or {}
     for key in res:
         if key == 'model_map':
             for k, v in custom.get(key, {}).items():
                 res[key][k] = full_spec(v)
-        else:
+        elif key == "renderer_map":
             res[key].update(custom.get(key, {}))
+        elif key == "extensions":
+            res[key].extend(custom.get(key, []))
```

Comment:
This will only allow adding to the defaults. Is this intended - i.e. could it be that a user explicitly doesn't want any of the default extensions?

Reply:
I think it would make sense to limit the default extensions to those that are actually needed to render whatever default renderers spit out (and maybe a short list of cross-platform well-supported things like `fenced_code` and `tables`). E.g. `md_in_html` or `footnotes` (and possibly `attr_list`) might be a bit too much.

Maybe this list could serve as some guideline: https://github.com/Crissov/CommonMark/blob/extensions/extensions/Desirable%20and%20popular%20Commonmark%20extensions.md

See also https://talk.commonmark.org/t/extension-terminology-and-rules/1233

Reply:
We could also take the stance that pandoc is a key tool to make CLDF markdown useful, thus limit inclusion of default extensions to ones available for pandoc as well (see https://pandoc.org/MANUAL.html#pandocs-markdown). I.e. we shouldn't encourage people to tailor CLDF markdown to a particular renderer (and if we do, that renderer should be pandoc rather than the clld-markdown-plugin). The main idea of CLDF is that datasets will outlive web apps - after all.

Reply:
> This will only allow adding to the defaults. Is this intended - i.e. could it be that a user explicitly doesn't want any of the default extensions?

I found it hard to come up with specific cases for that scenario, but I guess it could be -- e.g. if they use some non-standard custom extension that would interfere with a default.

Would a reasonable solution be to overwrite the defaults by using

`config.registry.settings["clld_markdown_plugin"]["extensions"] = [OnlyMyCustomExtension()]`

after `config.include("clld_markdown_plugin")`? It does work :)

I think if a desired output is LaTeX (for printing), then pandoc conformity makes sense.